src/transformStateWithClones.js

```diff
@@ -7,7 +7,36 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const newState = Object.assign({}, state);
+  const arrReturn = [];
+
+  for (const i of actions) {
+    if (i.type === 'clear') {
+      for (const key in newState) {
+        delete newState[key];
+      }
+    }
+
+    if (i.type === 'removeProperties') {
+      for (const rem of i.keysToRemove) {
+        for (const key in newState) {
+          if (key === rem) {
+            delete newState[key];
+          }
+        }
+      }
```

Comment:
It could be rewritten soooo much simpler _(by using forEach loop)_

```suggestion
      action.keysToRemove.forEach(key => delete state[key]); // action == i, in your case
```